app/src/main/java/fr/free/nrw/commons/bookmarks/BookmarkListRootFragment.java

```diff
@@ -172,15 +172,21 @@ public void viewPagerNotifyDataSetChanged() {
   }
 
   public void backPressed() {
-    if (mediaDetails.isVisible()) {
-      // todo add get list fragment
-      ((BookmarkFragment)getParentFragment()).tabLayout.setVisibility(View.VISIBLE);
-      removeFragment(mediaDetails);
-      setFragment(listFragment, mediaDetails);
-      ((MainActivity)getActivity()).showTabs();
+    //check medaiDetailPage fragment is not null then we check medaiDetail.is Visible or not to avoid NullPointerException
+    if (mediaDetails != null) {
+       if (mediaDetails.isVisible()) {
+        // todo add get list fragment
+        ((BookmarkFragment) getParentFragment()).tabLayout.setVisibility(View.VISIBLE);
+        removeFragment(mediaDetails);
+        setFragment(listFragment, mediaDetails);
+        ((MainActivity) getActivity()).showTabs();
```

Comment:
All code paths end with `((MainActivity) getActivity()).showTabs();` - we could move that to the end to reduce code duplication